sklearn/ensemble/tests/test_stacking_classifier.py

```diff
@@ -0,0 +1,217 @@
+from sklearn import datasets
+from sklearn.datasets import make_multilabel_classification
+from sklearn.svm import SVC
+from sklearn.multiclass import OneVsRestClassifier
+import numpy as np
+from sklearn.neighbors import KNeighborsClassifier
+
+
+# Load the iris dataset and randomly permute it
+iris = datasets.load_iris()
+X, y = iris.data[:, 1:3], iris.target
+
+
+def test_estimator_init():
+    clf = LogisticRegression(random_state=1)
+    eclf = StackingClassifier(estimators=[], meta_estimator=clf)
+    msg = ("Invalid `estimators` attribute %s, `estimators` "
+           "should be a list of (string, estimator) tuples.") % (type([]))
+    assert_raises(AttributeError, eclf.fit, X, y)
+
+    eclf = StackingClassifier(estimators=[('lr', clf)], meta_estimator=None)
+    msg = ("Invalid `meta_estimator` attribute %s, "
+           "`meta_estimator` should be a classifier.") % (type(None))
+    assert_raise_message(AttributeError, msg, eclf.fit, X, y)
+
+    clf_no_proba = SVC()
+    eclf = StackingClassifier(estimators=[('SVC', clf_no_proba)],
+                              meta_estimator=clf, method='predict_proba')
+    msg = ("Underlying estimator SVC <class 'sklearn.svm.classes.SVC'> "
+           "does not support predict_proba.")
+    assert_raise_message(ValueError, msg, eclf.fit, X, y)
+
+
+def test_notfitted():
+    eclf = StackingClassifier(estimators=[('lr1', LogisticRegression()),
+                                          ('lr2', LogisticRegression())],
+                              meta_estimator=LogisticRegression())
+    msg = ("This StackingClassifier instance is not fitted yet. Call \'fit\'"
+           " with appropriate arguments before using this method.")
+    assert_raise_message(NotFittedError, msg, eclf.predict, X)
+
+
+def test_multilabel():
+    """Check if error is raised for multilabel classification."""
+    X, y = make_multilabel_classification(n_classes=2, n_labels=1,
+                                          allow_unlabeled=False,
+                                          random_state=123)
+    clf = OneVsRestClassifier(SVC(kernel='linear'))
+    clf_meta = OneVsRestClassifier(SVC(kernel='linear'))
+
+    eclf = StackingClassifier(estimators=[('ovr1', clf)],
+                              meta_estimator=clf_meta,
+                              method='decision_function')
+
+    try:
+        eclf.fit(X, y)
+    except NotImplementedError:
+        return
+
+
+def test_cv():
+    """Test cross-validation option cv=1"""
+    clf1 = LogisticRegression(random_state=1)
+    clfm = GaussianNB()
+    eclf = StackingClassifier(estimators=[('lr', clf1)], cv=1,
+                              meta_estimator=clfm).fit(X, y)
+    s = clf1.fit(X, y).predict_proba(X)[:, 1:]
+    assert_array_equal(clfm.fit(s, y).predict(s), eclf.predict(X))
+    assert_array_equal(clfm.fit(s, y).predict_proba(s),
+                       eclf.predict_proba(X))
+
+
+def test_gridsearch():
+    """Check GridSearch support."""
```

Comment:
with nosetests, docstrings in test functions make test transcripts harder to read. make this a comment instead.